src/main/groovy/com/ullink/SourceLinkIndexing.groovy

```diff
@@ -42,6 +42,10 @@ class SourceLinkIndexing extends ConventionTask {
         }
     }
 
+    def findRepoRootDir(File dir) {
+        new File((File) dir, '.git').exists() ? dir : findRepoRootDir(dir.getParentFile())
```

Comment:
cast shouldn't be needed, also might be worth putting a limit to the recursion :)